File: AccountingDAL/Model/Plans/PlanSpending.cs
using System.ComponentModel.DataAnnotations.Schema;
using AccountingDAL.Model.Dictionaries;

namespace AccountingDAL.Model.Plans
{
    /// <summary>
    /// Планируемые траты
    /// </summary>
    public class PlanSpending : ModelElementBase
    {
        /// <summary>
        /// Идентификатор категории
        /// </summary>
        [ForeignKey(nameof(Category))]
        public Guid CategoryID { get; set; } = Guid.Empty;

        /// <summary>
        /// Категория
        /// </summary>
        public Category Category { get; set; }

        /// <summary>
        /// Планируемая сумма
        /// </summary>
        public float Sum { get; set; }

        /// <summary>
        /// Описание
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Идентификатор плана
        /// </summary>
        [ForeignKey(nameof(Plan))]
        public Guid PlanID { get; set; }

        /// <summary>
        /// План
        /// </summary>
        public Plan Plan { get; set; }
    }
}

File: Accounting_frontend/src/plan/model/plan-spending.ts
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { EMPTY_GUID } from 'src/const';
import { Category } from 'src/dictionaries/category/model/category';
import * as uuid from 'uuid';

/**
 * Планируемые траты
 */
export interface PlanSpending {
  /**
   * Идентификатор
   */
  id: string;

  /**
   * Идентификатор категории
   */
  categoryID: string;

  /**
   * Категория
   */
  category: Category | null;

  /**
   * Сумма
   */
  sum: number;

  /**
   * Описание
   */
  description: string;

  /**
   * Идентификатор плана
   */
  planID: string;
}

export function planSpendingDefault(): PlanSpending {
  return {
    id: uuid.v4(),
    categoryID: '',
    category: null,
    sum: 0,
    description: '',
    planID: EMPTY_GUID,
  };
}

export function planSpendingFormGroup(): FormGroup {
  return new FormGroup({
    id: new FormControl<string>(EMPTY_GUID),
    categoryID: new FormControl<string>(''),
    category: new FormControl<Category | null>(null, [Validators.required]),
    sum: new FormControl<number>(0),
    description: new FormControl<string>(''),
    planID: new FormControl<string>(''),
  });
}
